pkg/controller/registry/registry_controller_test.go

```diff
@@ -18,70 +18,290 @@
+			t.Logf("Error updating registry %v", err)
+		}
+
+		err = c.Delete(context.TODO(), registry)
+		if err != nil {
+			t.Logf("Error deleting registry %v", err)
+		}
+	}
+
+	secretName := types.NamespacedName{
+		Name:      "foo-ca-crt",
+		Namespace: metav1.NamespaceSystem,
+	}
+	secret := &corev1.Secret{}
+	err = c.Get(context.TODO(), secretName, secret)
+	//Get may fail if registry wasn't created, so ignore
+	if err == nil {
+		err = c.Delete(context.TODO(), secret)
+		if err != nil {
+			t.Logf("Error deleting secret %v", err)
+		}
+	}
+
+	crdName := "registries.kubic.opensuse.org"
+	crdClient, _ := clientset.NewForConfig(mgr.GetConfig())
+
+	err = crdClient.ApiextensionsV1beta1().
+		CustomResourceDefinitions().
+		Delete(crdName, &metav1.DeleteOptions{})
+
+	if err != nil {
+		t.Logf("Error deleting Registry CRD: %v", err)
+	}
+
+	cleanupJob(t, c, jobKey)
+
+	cleanupJob(t, c, checkJobKey)
+}
+
+func createCertificateCheckingJob(t *testing.T, c client.Client, regName string) (*batchv1.Job, error) {
+
+	job := &batchv1.Job{}
+
+	registry := &kubicv1beta1.Registry{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: regName}, registry)
+	if err != nil {
+		return job, err
+	}
+
+	registryAddress := kubicutil.SafeID(registry.Spec.HostPort)
+	secret, err := registry.GetCertificateSecret(c)
+	if err != nil {
+		return job, err
+	}
+
+	nodes, err := getAllNodes(c)
+	if err != nil {
+		return job, err
+	}
+	numNodes := len(nodes)
+
+	dockerDstDir := filepath.Join(dockerCertsDir, registry.Spec.HostPort)
+	podmanDstDir := filepath.Join(podmanCertsDir, registry.Spec.HostPort)
+
+	cmdTemplate := "if [[ ! -f '%s/ca.crt' ]]; then echo 'cert not found at %s'; exit 1; fi "
+	commands := []string{
+		fmt.Sprintf(cmdTemplate, dockerDstDir, dockerDstDir),
+		fmt.Sprintf(cmdTemplate, podmanDstDir, podmanDstDir),
+	}
+
+	job, err = getRunnerJobWithSecrets(&runnerWithSecrets{
+		Commands:     []string{strings.Join(commands, " ; ")},
+		JobName:      checkJobKey.Name,
+		NumNodes:     int32(numNodes),
+		JobNamespace: metav1.NamespaceSystem,
+		Secrets:      map[string]*corev1.Secret{},
+		Labels: map[string]string{
+			jobInstallLabelHostPort: registryAddress,
+			jobInstallLabelHash:     getSecretHash(secret),
+		},
+		HostPaths: []string{
+			"/etc/docker",
+			"/etc/containers",
+		},
+		AntiAffinity: map[string]string{
+			jobInstallLabelHostPort: registryAddress,
+		},
+	})
+
+	return job, err
+}
+
+func waitRegistryCreated(c client.Client, registryName string, timeout time.Duration) error {
+
+	registry := &kubicv1beta1.Registry{}
+	var err error
+	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		err = c.Get(context.TODO(), types.NamespacedName{Name: registryName}, registry)
+
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+
+		return true, nil
+	})
+	return err
+}
+
+func waitRegistryUpdates(c client.Client, registryName string, expected int32, timeout time.Duration) (int32, error) {
```

Comment:
```golang
func waitRegistryUpdates(c client.Client, registryName string, expected int32, timeout time.Duration) (int32, error) {
	registry := &kubicv1beta1.Registry{}
	numNodes := int32(-1)
	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
		err := c.Get(context.TODO(), types.NamespacedName{Name: registryName}, registry)
		if err != nil &&  !apierrors.IsNotFound(err) {
			return false, err
		}
                numNodes = int32(registry.Status.Certificate.NumNodes)
		return (numNodes == expected), nil
	})
	return numNodes, nil
}
```

Reply:
i think we could maybe simplify like this, i didn't tried

Reply:
The code you propose brakes the intended behavior. The intention is to return the `err` catched inside the function passed to `Poll`  from the `waitRegistryUpdates`. In the code you propose,  `err` is missing.